helios-client/src/main/java/com/spotify/helios/common/descriptors/PortMapping.java

```diff
@@ -41,31 +46,46 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class PortMapping extends Descriptor {
 
+  public static final String WILDCARD_ADDRESS = "0.0.0.0";
   public static final String TCP = "tcp";
   public static final String UDP = "udp";
 
+  private final String ip;
   private final int internalPort;
   private final Integer externalPort;
   private final String protocol;
 
-  public PortMapping(@JsonProperty("internalPort") final int internalPort,
-                     @JsonProperty("externalPort") final Integer externalPort,
-                     @JsonProperty("protocol") final String protocol) {
+  public PortMapping(@JsonProperty("ip") @Nullable final String ip,
```

Comment:
Can only see `PortMapping(final int internalPort, final Integer externalPort)` and `PortMapping(final int internalPort)` - but on second thought it seems unlikely that is used apart from json parsing